runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -50,12 +50,12 @@ end
 
 ---@private
 local function line_byte_from_position(lines, lnum, col, offset_encoding)
-  if not lines or offset_encoding == "utf-8" then
+  if not lines then
```

Comment:
I don't see why you would add additional indirection through a function call here

Reply:
I could revert this if you prefer. I just found it a bit cleaner since we have an LSP-specific function now.

Reply:
Oh I see what you mean, yeah there's no reason to not optimize this, will fix.

Reply:
+1 for reverting the changes in this file. Keeping the scope of the PR limited makes it easier to review.

Reply:
Reverted.